src/scheduler/gc_work.rs

```diff
@@ -277,7 +279,8 @@ impl<E: ProcessEdgesWork> ScanStackRoots<E> {
 impl<E: ProcessEdgesWork> GCWork<E::VM> for ScanStackRoots<E> {
     fn do_work(&mut self, worker: &mut GCWorker<E::VM>, mmtk: &'static MMTK<E::VM>) {
         trace!("ScanStackRoots");
-        <E::VM as VMBinding>::VMScanning::scan_thread_roots::<E>();
+        let factory = ProcessEdgesWorkRootsWorkFactory::<E>::new_boxed(mmtk);
```

Comment:
Yes.  Eventually, it will be the plan's responsibility to create the factory it needs.  The plan may even create a different factory for each different kind of GC (such as nursery vs full-heap, as well as concurrent marking).

I also plan to refactor the `ProcessEdgesWork`, but not in this PR.  This PR only changes the VM-facing API.